rfcs/43-many-worlds.md

```diff
@@ -0,0 +1,522 @@
+
+### WorldLabels
+
+Each world has its own label, which must implement the `WorldLabel` trait in the same fashion as the system labels you're already familiar with.
+
+By default, three worlds are added to your app: `CoreWorld::Global`, `CoreWorld::Main` and `CoreWorld::Rendering`.
+By convention, the main world should contain the vast majority of your logic, while the global world should only contain global resources (and other read-only data) that need to be accessed across worlds.
+
+### Global resources
+
+Each app has its own collection of global resources, which can be viewed (but not modified) in each of the worlds.
+This allows for convenient, non-blocking sharing of things like asset collections, settings and input events.
+
+If needed, you can get a read-only view into the entire global world using the `GlobalWorld` system parameter.
+This is non-blocking, as the global world cannot be modified in any fashion while other schedules are active.
+
+### Example: parallel simulations
+
+Suppose you're attempting to run a complex scientific simulation in parallel, and want to slightly vary the starting conditions or rules.
+Because these simulations don't need to interact in any meaningful way, this is an ideal situation in which to use multiple worlds.
+
+```rust
+use bevy::prelude::*;
+
+#[derive(WorldLabel)]
+struct SimulationWorld(usize);
+
+
+// These systems will be run in each of ours imulation worlds
+struct SimulationLogicPlugin;
+impl Plugin for SimulationLogicPlugin{
+   fn build(self, app: &mut App){
+      app
+         .add_system(births_system)
+         .add_system(deaths_system)
+         .add_system(migration_system)
+         // This system will send events to CoreWorld::Global
+         // so we can collect the simulation state for reporting
+         .add_system(report_status_system);
+   }
+}
+
+fn main(){
+
+   const N_WORLDS = 10;
+   let birth_rates: Vec<BirthRate> = BirthRate::random(10);
+   let death_rates: Vec<DeathRate> = DeathRate::random(10);
+
+   let mut app = App::new();
+   // These system is added to the global world's schedule,
+   // which will run after all of our simulation world
+   app.add_plugins(DefaultPlugins.to_world(CoreWorld::Global))
+   // This system will collect information send by events from each
+   // report_status_system in our parallel worlds,
+   // allowing us to watch the simulation progress over time in our window
+   app.add_system(plot_simulations_system.to_world(CoreWorld::Global));
+
+   for i in 0..N_WORLDS {
+      // Create a new world
+      app.spawn_world(SimulationWorld(i))
+         // Sets the currently active world for the App,
+         // which controls the schedule to which systems are added
+         // and where resources are inserted
+         .set_world(SimulationWorld(i))
+         // This ensures basic functionality like `Time` functions properly
+         // for each of our worlds
+         .add_plugins(MinimalPlugins)
```

Comment:
Would this even be necessary? I'm assuming something like `Time` would be in a global in such an architecture.

Reply:
Hmm. I think you're right.